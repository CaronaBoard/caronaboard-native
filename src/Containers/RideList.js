
import React, { Component } from 'react'
import { ListView, StyleSheet } from 'react-native'
import Ride from '../Components/Ride'

export default class RideList extends Component {
  constructor (props) {
    super(props)
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id})
    this.state = {
      dataSource: ds.cloneWithRows(props.rides)
    }
  }

  render () {
    return (
      <ListView
        style={styles.container}
        dataSource={this.state.dataSource}
        renderRow={(ride) => <Ride ride={ride} />}
      />
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 20
  }
})
