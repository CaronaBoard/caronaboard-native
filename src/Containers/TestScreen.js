import React, { Component } from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'

import { testAction } from '../Redux/Actions/TestActions'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './Styles/SignInScreenStyles'

export class TestScreen extends Component {
  render () {
    const { testCounter, increaseCounter } = this.props
    return (
      <View style={Styles.flexible}>
        <View style={Styles.container}>
          <View>
            <RkText style={Styles.title}>
              {testCounter}
            </RkText>
          </View>
          <View>
            <RkButton innerStyle={[Styles.buttonFontSize]}
              rkType='circle outline medium'
              onPress={() => increaseCounter()}>
              UPDATE STATE
            </RkButton>
          </View>
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    testCounter: state.test.counter
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    increaseCounter: () => dispatch(testAction())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TestScreen)
