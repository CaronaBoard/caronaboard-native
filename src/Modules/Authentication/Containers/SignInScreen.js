import Styles from './Styles/SignInScreenStyles'
import React, { Component } from 'react'
import { View } from 'react-native'
import { RkText } from 'react-native-ui-kitten'
import { Button, TextInput } from '../../Shared/Components'
import { signInFirebase } from '../../../Redux/Actions'
import { connect } from 'react-redux'

export class SignInScreen extends Component {
  constructor (props) {
    super()
    this.state = {
      email: '',
      password: '',
      loading: false
    }
  }

  onButtonSubmit = () => {
    this.props.signIn(this.state.email, this.state.password).then(this.onLoginSuccess)
  }

  onLoginSuccess = (userData) => {
    // TODO navigate to ridelist?
  }

  render () {
    return (
      <View style={Styles.flexible}>
        <View style={Styles.container}>
          <RkText style={Styles.title}>
            SIGN IN
          </RkText>
          <View style={Styles.credentialsContainer}>
            <TextInput label='EMAIL ADDRESS' onChangeText={(email) => { this.setState({email}) }} />
            <TextInput label='PASSWORD' onChangeText={(password) => { this.setState({password}) }} />
          </View>
          <Button text='SIGN IN' onPress={this.onButtonSubmit} />
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    userData: state.auth.userData
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    signIn: (e, p) => dispatch(signInFirebase(e, p))
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(SignInScreen)
