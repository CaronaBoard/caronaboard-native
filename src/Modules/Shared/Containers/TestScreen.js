import React, { Component } from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'

import { testAction } from '../../../Redux/Actions/TestActions'
import { RkText } from 'react-native-ui-kitten'
import { Button } from '../../Shared/Components'
import Styles from '../../Authentication/Containers/Styles/SignInScreenStyles'

export class TestScreen extends Component {
  render () {
    const { testCounter, increaseCounter } = this.props
    return (
      <View style={Styles.flexible}>
        <View style={Styles.container}>
          <View>
            <RkText style={Styles.title}>
              {testCounter}
            </RkText>
          </View>
          <Button text={'UPDATE STATE'} onPress={() => increaseCounter()} />
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    testCounter: state.test.counter
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    increaseCounter: () => dispatch(testAction())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TestScreen)
