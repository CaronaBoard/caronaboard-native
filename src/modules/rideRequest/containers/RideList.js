
import React, { Component } from 'react'
import { ListView, StyleSheet } from 'react-native'
import { fetchAllRides } from '../../../redux/actions'
import { connect } from 'react-redux'

import { Ride } from '../components/Ride'

export class RideList extends Component {
  constructor (props) {
    super(props)
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id})
    this.state = {
      dataSource: ds.cloneWithRows(this.props.rides)
    }
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.rides.length !== this.props.rides.length) {
      const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id})
      this.setState({
        dataSource: ds.cloneWithRows(nextProps.rides)
      })
    }
  }

  componentDidMount () {
    this.props.fetchRides()
  }

  render () {
    return (
      <ListView
        style={styles.container}
        dataSource={this.state.dataSource}
        renderRow={(ride) => <Ride ride={ride} navigator={this.props.navigator} />}
        enableEmptySections
      />
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 20
  }
})

const mapStateToProps = (state) => {
  return {
    rides: state.rideOffer.rides
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRides: () => dispatch(fetchAllRides())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(RideList)
