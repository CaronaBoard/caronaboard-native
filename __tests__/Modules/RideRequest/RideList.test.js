import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { RideList } from '../../../src/Modules/RideRequest/Containers/RideList'

const props = {
  rides: [{
    'area': 'Aeroporto',
    'days': 'Seg a Sex',
    'destination': 'Rodoviaria',
    'flexible': true,
    'formUrl': 'https://goo.gl/forms/',
    'hours': '19h',
    'name': 'Hugh Jackman',
    'origin': 'Tecnopuc',
    'id': 1
  }],
  fetchRides: jest.fn()
}

describe('<RideList />', () => {
  const wrapper = shallow(<RideList {...props} />)

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
