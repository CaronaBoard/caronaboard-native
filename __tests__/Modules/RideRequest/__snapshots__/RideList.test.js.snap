// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RideList /> Should have a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ListViewMock
    dataSource={
        ListViewDataSource {
            "items": 1,
          }
    }
    enableEmptySections={true}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    style={
        Object {
            "flex": 1,
            "paddingTop": 20,
          }
    }
/>,
  "nodes": Array [
    <ListViewMock
      dataSource={
            ListViewDataSource {
                  "items": 1,
                }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      style={
            Object {
                  "flex": 1,
                  "paddingTop": 20,
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RideList
        fetchRides={[Function]}
        rides={
                Array [
                        Object {
                          "area": "Aeroporto",
                          "days": "Seg a Sex",
                          "destination": "Rodoviaria",
                          "hours": "19h",
                          "origin": "Tecnopuc",
                          "profile": Object {
                            "contact": Object {
                              "kind": "Whatsapp",
                              "value": "5198269999",
                            },
                            "name": "Eduardo",
                          },
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RideList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "fetchRides": [Function],
          "rides": Array [
            Object {
              "area": "Aeroporto",
              "days": "Seg a Sex",
              "destination": "Rodoviaria",
              "hours": "19h",
              "origin": "Tecnopuc",
              "profile": Object {
                "contact": Object {
                  "kind": "Whatsapp",
                  "value": "5198269999",
                },
                "name": "Eduardo",
              },
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "dataSource": ListViewDataSource {
            "items": 1,
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ListViewMock
          dataSource={
                    ListViewDataSource {
                              "items": 1,
                            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
                    Object {
                              "flex": 1,
                              "paddingTop": 20,
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <ListViewMock
          dataSource={
                    ListViewDataSource {
                              "items": 1,
                            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
                    Object {
                              "flex": 1,
                              "paddingTop": 20,
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RideList
    fetchRides={[Function]}
    rides={
        Array [
            Object {
              "area": "Aeroporto",
              "days": "Seg a Sex",
              "destination": "Rodoviaria",
              "hours": "19h",
              "origin": "Tecnopuc",
              "profile": Object {
                "contact": Object {
                  "kind": "Whatsapp",
                  "value": "5198269999",
                },
                "name": "Eduardo",
              },
            },
          ]
    }
/>,
}
`;
